src/histolab/tiler.py

```diff
@@ -113,6 +114,18 @@ def locate_tiles(
 
     # ------- implementation helpers -------
 
+    def _fix_tile_size_if_mpp(self, slide):
+        """
+        Set tile size either to requested level or if MPP is requested,
+        set tile size relative to base mpp of slide instead.
+        """
+        if self.mpp is None:
+            return
+        sf = self.mpp / slide.base_mpp
+        self.tile_size = tuple(int(j * sf) for j in self.tile_size)
+        if hasattr(self, "pixel_overlap"):
+            self.pixel_overlap = int(self.pixel_overlap * sf)
```

Comment:
I have some concerns on this method:
1. You are setting a Protocol variable in here that basically has some security vulnerabilities because we change the object info
2. You are changing also a shadow name here `pixel_overlap` using the trick of the hasattr... that smells to me. Pixel_overlap doesn't exists in the scope of the `Tiler` object and so it should be set in each Tiler subclass.

I would suggest to remove this method frome here a use setter and getter in each class you need this. If this is not clear I'll try to explain it better. Just let me know

Reply:
OK, now the ``Tiler`` protocol only has a getter and does not use hasattr. Setters are used for each of ``GridTiler`` and ``RandomTiler`` to update ``tile_size`` and ``pixel_overlap`` as needed. Test coverage no also ensures ``pixel_overlap`` is successfully updated if ``mpp`` is set.